third_party/nvfuser/csrc/python_frontend/fusion_definition.cpp

```diff
@@ -100,42 +88,37 @@ void FusionDefinition::finalizeDefinition() {
   FUSER_PERF_SCOPE("FusionDefinition::finalizeDefinition");
   auto cache_entry = fusionCache()->queryChildren(end_record_.get());
   if (!cache_entry.has_value()) {
-    if (nvfuser::isDebugDumpEnabled(
-            nvfuser::DebugDumpOption::PythonFrontendDebug)) {
+    if (isDebugDumpEnabled(DebugDumpOption::PythonFrontendDebug)) {
       std::cout << "\nFusionDefinition: Terminal Node not found.\n";
     }
     fusion_id_ = fusionCache()->createChild(end_record_.get());
-    TORCH_CHECK(fusion_id_.has_value(), "Invalid fusion id!");
+    TORCH_CHECK(id().has_value(), "Invalid fusion id!");
     fusionCache()->traverseTrie(end_record_.get());
 
-    if (nvfuser::isDebugDumpEnabled(
-            nvfuser::DebugDumpOption::PythonDefinition)) {
+    if (isDebugDumpEnabled(DebugDumpOption::PythonDefinition)) {
       print(std::cout);
     }
 
-    fusion_ = preschedFusion();
-    buildFusionIr();
+    buildFusionIr(preschedFusion());
 
-    if (nvfuser::isDebugDumpEnabled(
-            nvfuser::DebugDumpOption::FusionIrPresched)) {
+    if (isDebugDumpEnabled(DebugDumpOption::FusionIrPresched)) {
       printIr();
     }
   } else {
-    if (nvfuser::isDebugDumpEnabled(
-            nvfuser::DebugDumpOption::PythonFrontendDebug)) {
+    if (isDebugDumpEnabled(DebugDumpOption::PythonFrontendDebug)) {
       std::cout << "\nFusionDefinition: Terminal Node found!\n";
     }
+    auto new_fusion_id = c10::optional<size_t>(cache_entry.value()->fusion_id);
```

Comment:
I get a compilation error as `new_fusion_id` isn't used. Looks like the line below is almost exactly the same except it stores to `fusion_id_`. Can this line be just removed?

Reply:
Yes. It should be removed.

Reply:
https://github.com/csarofeen/pytorch/pull/2519